synapse/storage/databases/main/receipts.py

```diff
@@ -843,8 +830,8 @@ def _insert_graph_receipt_txn(
         else:
             keyvalues["thread_id"] = thread_id
 
-        self.db_pool.simple_upsert_txn(
-            txn,
+        await self.db_pool.simple_upsert(
```

Comment:
I think you still want to use the `_txn` version of this and keep everything inside the same transaction?

Reply:
I thought we inserted both the linearized and graph receipts in the same transaction?